### PR TITLE
Add joy example for circle row

### PR DESCRIPTION
**Pull request by kushalpokharel**
I have added the example for a row of 3 circles drawn using `Joy.Shape` as required by [this](https://github.com/Sudha247/ocaml-joy/issues/28) issue.

### CODE REVIEW

File: test/test_circle_row.ml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+let () =
+  let (dim_x, dim_y) = (1000,1000) in
+  set_dimensions dim_x dim_y;
+  init ();
+  let spacing = 120 in 
+  let circles = row_of_circles dim_x dim_y spacing in
+  show circles;
+  ignore (read_line ());
+  close()
```
Comment (nikochiko) [inline test/test_circle_row.ml:22]: @kushalpokharel this example can be made simpler with the recent `translate` transformation that was added. Can you please use that instead?

Here is an example of its usage -- https://github.com/Sudha247/ocaml-joy/blob/main/examples/translate_circle.ml

Some other pointers:
- please stick with the default for dimensions
- i don't think we need another function, or to use variables
- please don't use references (like `circles` in this example)

as much as possible, we want our examples to be inviting to newbies. simplicity will go a long way in that!

**Review by nikochiko — CHANGES_REQUESTED** [review]
@kushalpokharel nice work! :)

i have added some comments on making the example simpler.

File: test/test_circle_row.ml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+open Joy.Shape
+
+let run ()=
+  init ();
+  let base_circle = circle 50 in
+  let circle1 = translate 150 0 base_circle in
+  let circle2 = translate (-150) 0 base_circle in
+  show [base_circle; circle1; circle2];
+  ignore (read_line ());
```
Comment (nikochiko) [inline test/test_circle_row.ml:9]: this is not needed, it's already there in `close ()`

**Review by nikochiko — COMMENTED** [review]
This looks really nice! Great work @kushalpokharel

File: test/test_circle_row.ml (hunk 2 above)
Comment (kushalpokharel) [reply test/test_circle_row.ml]: Hi, I have removed this. Please review it

Comment (nikochiko) [reply test/test_circle_row.ml]: Great job! 🎉

**Comment by nikochiko** [comment]
I've made some minor changes to the existing code. LGTM!

**Review by nikochiko — CHANGES_REQUESTED** [review]
Oops, I didn't notice that the issue asked to create an example and not a test.

@kushalpokharel the changes look good to me. Can you please move the `test_circle_row.ml` file to `examples/circle_row_joy.ml` and remove its import from the `test_all.ml` file?

@Sudha247 how do you think we should organise files in `examples/` directory? -- since we have examples for both graphics and joy library.

**Comment by kushalpokharel** [comment]
@nikochiko , I hope this resolves all the issues

**Comment by nikochiko** [comment]
@kushalpokharel this looks good to me. Can you please resolve conflicts with main so this can be merged?

**Review by nikochiko — APPROVED** [review]
Thanks! @kushalpokharel nice work!

**Pull request merged**